### PR TITLE
Notifications test case

### PR DESCRIPTION
**Pull request by thelonelyvulpes**

### CODE REVIEW

**Review by injectives — APPROVED** [review]

**Comment by injectives** [comment]
We might need to update backends to permit `None` access mode.

File: tests/stub/notifications_config/test_session_notifications_config.py [hunk 1]
```diff
@@ -58,6 +60,33 @@ def test_session_overrides_driver_config_on_read_tx(self):
...
+                    )
+                    script_vars = {
+                        "#NOTIS#": '"notifications_minimum_severity"'
+                                   ': "WARNING", ',
+                        "#HOST#": host
+                    }
+                    self._server.start(script_path, vars_=script_vars)
+                    driver = Driver(self._backend, uri, self._auth)
+                    session = driver.session(None,
+                                             notifications_min_severity="WARNING")  # noqa: E501
```
Comment (robsdedude) [inline tests/stub/notifications_config/test_session_notifications_config.py]:
```suggestion
                    session = driver.session(
                        None, notifications_min_severity="WARNING"
                    )
```

File: tests/stub/notifications_config/test_session_notifications_config.py [hunk 2]
```diff
@@ -58,6 +60,33 @@ def test_session_overrides_driver_config_on_read_tx(self):
                                    script, self.driver_params)
             self._server.reset()
 
+    def test_session_config_with_database_not_specified(self):
+        uris = [self._uri, self._uri.replace("bolt", "neo4j")]
+        for uri in uris:
+            with self.subTest(name=uri):
```
Comment (robsdedude) [inline tests/stub/notifications_config/test_session_notifications_config.py]:
```suggestion
            with self.subTest(uri=uri):
```
nit

File: tests/stub/notifications_config/test_session_notifications_config.py [hunk 3]
```diff
@@ -58,6 +60,33 @@ def test_session_overrides_driver_config_on_read_tx(self):
...
+                try:
+                    host = os.environ.get("TEST_STUB_HOST",
+                                          "127.0.0.1") + ":" + str(self._port)
```
Comment (robsdedude) [inline tests/stub/notifications_config/test_session_notifications_config.py]: That's exactly `self._server.address`. No need to duplicate the logic.

File: tests/stub/notifications_config/test_session_notifications_config.py [hunk 4]
```diff
@@ -58,6 +60,33 @@ def test_session_overrides_driver_config_on_read_tx(self):
...
+                    driver = Driver(self._backend, uri, self._auth)
+                    session = driver.session(None,
+                                             notifications_min_severity="WARNING")  # noqa: E501
+                    cursor = session.run("RETURN 1 as n")
+                    cursor.consume()
+                    self._server.done()
+                finally:
+                    self._server.reset()
+                if driver is not None:
+                    driver.close()
```
Comment (robsdedude) [inline tests/stub/notifications_config/test_session_notifications_config.py]: was this supposed to be in the `finally` block?

**Review by robsdedude — CHANGES_REQUESTED** [review]

File: tests/stub/notifications_config/test_session_notifications_config.py [hunk 5]
```diff
@@ -58,6 +58,34 @@ def test_session_overrides_driver_config_on_read_tx(self):
...
+                    session = driver.session(
+                        None,
+                        notifications_min_severity="WARNING"
+                    )
+                    cursor = session.run("RETURN 1 as n")
+                    cursor.consume()
+                    self._server.done()
+                finally:
+                    self._server.reset()
+                    if driver is not None:
```
Comment (robsdedude) [inline tests/stub/notifications_config/test_session_notifications_config.py:87]: Note that when the script doesn't make it to line 76
```python
driver = Driver(self._backend, uri, self._auth)
```
`driver` will not be `None` but undefined => Exception. You'd have to define `driver = None` before the try to be 100% this works as intended.

**Review by robsdedude — APPROVED** [review]
:film_projector: LGTM except for another small comment

File: tests/stub/notifications_config/test_session_notifications_config.py [hunk 6]
```diff
@@ -58,6 +58,34 @@ def test_session_overrides_driver_config_on_read_tx(self):
...
+                                   ': "WARNING", ',
+                        "#HOST#": host
+                    }
+                    self._server.start(script_path, vars_=script_vars)
+                    driver = Driver(self._backend, uri, self._auth)
+                    session = driver.session(
+                        None,
+                        notifications_min_severity="WARNING"
+                    )
+                    cursor = session.run("RETURN 1 as n")
```
Comment (robsdedude) [inline tests/stub/notifications_config/test_session_notifications_config.py]:
```suggestion
                    cursor = session.run("RETURN 1 AS n")
```
nit (`AS` is a keyword) + adjust the script if accepted

**Comment by thelonelyvulpes** [comment]
Stale

**Pull request closed without merging**